common/src/View/MapDocument.cpp

```diff
@@ -1471,58 +1471,114 @@ namespace TrenchBroom {
             resetLock(nodesToReset);
         }
 
-        bool MapDocument::translateObjects(const vm::vec3& delta) {
-            const auto result = executeAndStore(TransformObjectsCommand::translate(delta, pref(Preferences::TextureLock)));
-            if (result->success()) {
-                m_repeatStack->push([=]() { this->translateObjects(delta); });
-                return true;
+        void MapDocument::swapNodeContents(const std::string& commandName, std::vector<std::tuple<Model::Node*, std::unique_ptr<Model::Node>>> nodesToSwap) {
+            executeAndStore(std::make_unique<SwapNodeContentsCommand>(commandName, std::move(nodesToSwap)));
+        }
+        
+        /**
+         * Clone each of the given nodes, apply the given lambda to the clone, and swap the contents of each original node with those of the modified clone.
+         *
+         * Returns true if the given lambda could be applied successfully to all clones and false otherwise. If the lambda fails for any clone, then no
+         * node contents will be swapped, and the original nodes remain unmo
+         */
+        template <typename L>
+        static bool applyAndSwap(MapDocument& document, const std::string& commandName, const std::vector<Model::Node*>& nodes, L lambda) {
```

Comment:
the usual idiom is a `L&& lambda` forwarding reference for the lambda param, right?

Reply:
These days I usually prefer taking by value and moving like here. This didn't work for the recursive lambdas in the other PR because we couldn't move the parameter, so it needed to be a const lvalue ref or a forwarding ref.